txpipe/twopoint_fourier.py

```diff
@@ -157,111 +156,128 @@ def load_maps(self):
-        for m in g1_maps + g2_maps + ngal_maps:
-            mask[m==healpy.UNSEEN] = 0
+        for (g1, g2, lw) in zip(g1_maps, g2_maps, lensing_weights):
+            lw[g1 == healpy.UNSEEN] = 0
+            lw[g2 == healpy.UNSEEN] = 0
+
+        # When running on the CosmoDC2 mocks I've found I need to flip
+        # both g1 and g2 in order to get both positive galaxy-galaxy lensing
+        # and shear-shear spectra.
+        if self.config['flip_g1']:
+            for g1 in g1_maps:
+                w = np.where(g1!=healpy.UNSEEN)
+                g1[w]*=-1
 
         if self.config['flip_g2']:
             for g2 in g2_maps:
                 w = np.where(g2!=healpy.UNSEEN)
                 g2[w]*=-1
 
         # TODO: load systematics maps here, once we are making them.
+        # maybe from Eli or Zilong's code
         syst_nc = None
         syst_wl = None
 
         map_file.close()
 
-        # Cut out any pixels below the threshold,
-        # zeroing out any pixels there
-        cut = mask < mask_threshold
-        mask[cut] = 0
+        # THe gnomonic (tangent plane) mapping versions need the map width
+        # I'm thinking of deleting support for this since it adds a whole
+        # set of options to maintain.
+        if pixel_scheme.name == 'gnomonic':
+            lx = np.radians(pixel_scheme.size_x)
+            ly = np.radians(pixel_scheme.size_y)
+
+        # For the lensing mask we optionall apodize the mask.
+        # TODO: include the masked object fraction in here.
+        # TODO: ask about including the depth map in here
+        if apod_size > 0:
+            if self.rank==0:
+                print(f"Apodizing clustering weights map with size {apod_size} deg and method {apod_type}")
+
+            # NaMaster has different functions for apodizing the two different maps
+            if pixel_scheme.name == 'gnomonic':
+                clustering_weight = nmt.mask_apodization_flat(clustering_weight, lx, ly, apod_size, apotype=apod_type)
+            elif pixel_scheme.name == 'healpix':
+                clustering_weight = nmt.mask_apodization(clustering_weight, apod_size, apotype=apod_type)
+            else:
+                raise ValueError(f"Pixelization scheme {pixel_scheme.name} not supported by NaMaster")
 
-        # We also apply this cut to the count maps,
-        # since otherwise the pixels below threshold would contaminate
-        # the mean calculation below.
+        # Set any unseen pixels to zero weight.
         for ng in ngal_maps:
-            ng[cut] = 0
+            clustering_weight[ng==healpy.UNSEEN] = 0
 
+        d_maps = []
+        for ng in ngal_maps:
+            # Convert the number count maps to overdensity maps.
+            # First compute the overall mean object count per bin.
+            # Maybe we should do this in the mapping code itself?
+            # mean clustering galaxies per pixel in this map
+            mu = ng[clustering_weight>0].mean()
```

Comment:
is this a weighted mean?

Reply:
Actually, what is `mu` exactly? Is it a map?
In principle each pixel should contain:
```
delta[i] = number[i] / (mean_number * mask[i]) - 1
```
where `number` is the number of objects in that pixel, `mask` is the effective area fraction of that pixel (is this what you're calling `clustering_weight`?) and `mean_number = np.sum(number * mask) / np.sum(mask)`.